plugins/Ecommerce/Columns/Revenue.php

```diff
@@ -27,6 +33,55 @@ public function getDbDiscriminator()
         return new Discriminator($this->dbTableName, 'idgoal', GoalManager::IDGOAL_ORDER);
     }
 
+
+    public function configureSegments(SegmentsList $segmentsList, DimensionSegmentFactory $dimensionSegmentFactory)
+    {
+        //new Segment revenue on order
+        $segment = new Segment();
+        $segment->setCategory($this->category);
+        $segment->setName(Piwik::translate('Ecommerce_OrderRevenue'));
+        $segment->setSegment('revenueOrder');
+        $segment->setSqlSegment('log_conversion.idvisit');
+        $segment->setSqlFilter(function ($valueToMatch, $sqlField, $matchType) {
+            return $this->getRevenueQuery($valueToMatch, $sqlField , $matchType, 0);
+        });
+        $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+
+        //new Segment revenue left in cart
+        $segment = new Segment();
+        $segment->setCategory($this->category);
+        $segment->setName(Piwik::translate('Ecommerce_RevenueLeftInCart'));
+        $segment->setSegment('revenueAbandonedCart');
+        $segment->setSqlSegment('log_conversion.idvisit');
+        $segment->setSqlFilter(function ($valueToMatch, $sqlField , $matchType) {
+           return $this->getRevenueQuery($valueToMatch, $sqlField , $matchType,-1);
+        });
+        $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+
+    }
+
+    /**
+     * revenue sql filter
+     * @param $valueToMatch
+     * @param $sqlField
+     * @param $matchType
+     * @param false $cart
+     * @return array
+     */
+    private function getRevenueQuery($valueToMatch, $sqlField , $matchType, $idGoal)
+    {
+        if ($matchType === SegmentExpression::MATCH_EQUAL) {
+            $matchType = "=";
+        }
```

Comment:
You should better handle all supported types here (like in the file I mentioned earlier) and throw an exception for not supported types. Otherwise it would throw an sql error if someone defines a segment like `revenueOrder!@5`